include/xtensor/xhistogram.hpp

```diff
@@ -0,0 +1,307 @@
+            {
+                count[i] /= ( (bin_edges[i+1]-bin_edges[i]) * N );
+            }
+        }
+
+        return count;
+    }
+
+    /**
+     * @ingroup basic_functions
+     * @brief Compute the histogram of a set of data.
+     *
+     * @param The data.
+     * @param The bin-edges.
+     * @param If true the resulting integral is normalized to 1. [default: false]
+     * @return An one-dimensional xarray<double>, length: bin_edges.size()-1.
+     */
+    template<class E1, class E2>
+    inline auto histogram(
+        const E1 &data, const E2 &bin_edges, bool density=false
+    )
+    {
+        using value_type = typename E1::value_type;
+
+        xt::xtensor<value_type,1> weights = xt::ones<value_type>({data.size()});
+
+        return histogram(data, bin_edges, weights, density);
+    }
+
+    /**
+     * @ingroup basic_functions
+     * @brief Compute the histogram of a set of data.
+     *
+     * @param The data.
+     * @param The number of bins. [default: 10]
+     * @param If true the resulting integral is normalized to 1. [default: false]
+     * @return An one-dimensional xarray<double>, length: bin_edges.size()-1.
+     */
+    template<class E1>
+    inline auto histogram(
+        const E1 &data, size_t bins=10, bool density=false
+    )
+    {
+        using value_type = typename E1::value_type;
+
+        xt::xtensor<value_type,1> weights = xt::ones<value_type>({data.size()});
+
+        auto bin_edges = histogram_bin_edges(data, weights, bins);
+
+        return histogram(data, bin_edges, weights, density);
+    }
+
+    /**
+     * @ingroup basic_functions
+     * @brief Compute the histogram of a set of data.
+     *
+     * @param The data.
+     * @param The number of bins.
+     * @param Weight factors corresponding to each data-point.
+     * @param If true the resulting integral is normalized to 1. [default: false]
+     * @return An one-dimensional xarray<double>, length: bin_edges.size()-1.
+     */
+    template<class E1, class E2>
+    inline auto histogram(
+        const E1 &data, size_t bins, const E2 &weights, bool density=false
+    )
+    {
+        auto bin_edges = histogram_bin_edges(data, weights, bins);
+
+        return histogram(data, bin_edges, weights, density);
+    }
+
+    /**
+     * @ingroup basic_functions
+     * @brief Compute the bin-edges of a histogram of a set of data using different algorithms.
+     *
+     * @param The data.
+     * @param Weight factors corresponding to each data-point.
+     * @param The lower-most edge.
+     * @param The upper-most edge.
+     * @param The number of bins. [default: 10]
+     * @param The type of algorithm to use. [default: "auto"]
+     * @return An one-dimensional xarray<double>, length: bins+1.
+     */
+    template<class E1, class E2, class E3>
+    inline auto histogram_bin_edges(const E1 &data, const E2 &weights,
+        E3 left, E3 right, size_t bins=10, const std::string &mode="auto"
```

Comment:
An enum class would be better than string for the algorithm to use.

Reply:
@JohanMabille Do you have a suggestion for a good name. My naive suggestion is

```cpp
enum class Histogram { Auto, Linspace, Uniform };
```

but I feel that there might exist better.

Reply:
Well I'm not really good at finding names, so usually I try to use explicit names (but that may be long): `histogram_mode`, `histogram_algorithm`, or shortened version (`histo_mode`, `histo_algorithm`, `histo_algo`)

Reply:
@JohanMabille Ok, I'll pick one. One 'issue' is still the use of capitals. In the class name they can be easily avoided. The same is not true for the items as `auto` is not allowed.